autopartitioning-eng (#10532)
ydb-platform/ydb: ydb/docs/en/core/yql/reference/yql-core/syntax/alter-topic.md

* `supported_codecs`: List of [codecs](../../../../concepts/topic#message-codec) supported by the topic. Value type: `String`.

{% endif %}

### Change autopartitioning strategies for the topic {#autopartitioning}

> The following command sets the [autopartitioning](../../../../concepts/topic#autopartitioning) strategy to `UP`:

Reviewer: ```suggestion
The following command sets the [autopartitioning](../../../concepts/topic.md#autopartitioning) strategy to `UP`:
```

```yql
ALTER TOPIC `my_topic` SET (
    min_active_partitions = 1,
    max_active_partitions = 5,

>     auto_partitioning_strategy = 'scale_up'

Reviewer: Two naming inconsistencies:

* above, we call these "modes", while the actual setting name is "strategy"
* above, we call this mode/strategy just "up", while the actual setting value is "scale up"

Author: 1. I'll change "mode" into the "strategy"? 
2. YQL will be fixed here, and after that both "scale up" & "up" will be possible, and we'll change the documentation after that (not in this PR).

);
```

> The following command pauses the topic [autopartitioning](../../../../concepts/topic#autopartitioning):

Reviewer: ```suggestion
The following command pauses the topic [autopartitioning](../../../concepts/topic.md#autopartitioning):
```

> The following command sets the [autopartitioning](../../../../concepts/topic#autopartitioning) strategy to `UP`:
> 
> ```yql
> ALTER TOPIC `my_topic` SET (
>     min_active_partitions = 1,
>     max_active_partitions = 5,
>     auto_partitioning_strategy = 'scale_up'
> );
> ```
> 
> The following command pauses the topic [autopartitioning](../../../../concepts/topic#autopartitioning):
> 
> ```yql
> ALTER TOPIC `my_topic` SET (
>     auto_partitioning_strategy = 'paused'
> );
> ```
> 
> The following command unpauses the topic [autopartitioning](../../../../concepts/topic#autopartitioning):

Reviewer: ```suggestion
The following command sets the [autopartitioning](../../../concepts/topic.md#autopartitioning) strategy to `UP`:

```yql
ALTER TOPIC `my_topic` SET (
    min_active_partitions = 1,
    max_active_partitions = 5,
    auto_partitioning_strategy = 'scale_up'
);
```

The following command pauses the topic [autopartitioning](../../../concepts/topic.md#autopartitioning):

```yql
ALTER TOPIC `my_topic` SET (
    auto_partitioning_strategy = 'paused'
);
```

The following command unpauses the topic [autopartitioning](../../../concepts/topic.md#autopartitioning):
```

> The following command unpauses the topic [autopartitioning](../../../../concepts/topic#autopartitioning):

Reviewer: ```suggestion
The following command unpauses the topic [autopartitioning](../../../concepts/topic.md#autopartitioning):
```
